### DISCUSSION TITLE
How do I import my backup (made on MongoDB) into my newly installed version with PosgreSQL?

### DISCUSSION THREAD

**Discussion by exander77** [Ylianst/MeshCentral#5773] [Q&A]
Pretty much summed in title.

**Comment by si458** [comment]
You can't as far as im aware because you can't migrate between databases, you have to stay on the same database

But the is a restore backup option in the Web ui, so I guess u could create a new server with new database, then try restoring ur backup zip from the web ui?

**Reply by exander77** [reply]
Yes, I want to restore a backup (which is the backup that is produced by backup feature). But I don't even see an option in UI to restore it. I thought this will be easy as I have daily backups turned on. That I will start a new app and import from backup and everything would work. I have clean up, I am admin, I don't see any option to restore anywhere.

**Reply by si458** [reply]
Hmm I will have to check tomorrow? The restore backup option might only be visible IF the database is nedb!?

In the mean time u could try exporting the database from the old server and importing into the new dB?

https://ylianst.github.io/MeshCentral/meshcentral/#database-export

https://ylianst.github.io/MeshCentral/meshcentral/#database-import

**Reply by exander77** [reply]
![image](https://github.com/Ylianst/MeshCentral/assets/2256039/ae356c34-e72d-47aa-9f1b-613b94f9eeda)

**Reply by exander77** [reply]
I currently only have daily backups for around 100 days. I hadn't thought I will need the original database for anything. :/

So I would need to install mongo, restore it, export it and import it? But how do I import it on mongo if it is only visible on NeDB?

**Reply by si458** [reply]
I think u would have to restore ur old database, setup meshcentral with ur old config.json, then run the database export from link above, then copy the export to ur new server with the new database, then run the database import from again the link above

The web ui backup/restore is different to the database export/import

But this will all be trial and error I'm afraid!

I'm still new to maintaining meshcentral and with so many hidden features im not aware of!

**Reply by exander77** [reply]
So the UI backup/restore is not what is in autobackups, how do I restore autobackup then? Do I use mongo restore?

**Reply by exander77** [reply]
I just tried mongorstore on the DB:
```
root@75566d3c2072:/data/backup# mongorestore mongodump-2024-02-06-14-22.archive 
2024-02-06T23:02:06.611+0000	Failed: file mongodump-2024-02-06-14-22.archive does not have .bson extension
2024-02-06T23:02:06.611+0000	0 document(s) restored successfully. 0 document(s) failed to restore.
root@75566d3c2072:/data/backup# mv mongodump-2024-02-06-14-22.archive mongodump-2024-02-06-14-22.archive.bson
root@75566d3c2072:/data/backup# mongorestore mongodump-2024-02-06-14-22.archive.bson 
2024-02-06T23:02:19.225+0000	checking for collection data in mongodump-2024-02-06-14-22.archive.bson
2024-02-06T23:02:19.255+0000	restoring test.mongodump-2024-02-06-14-22.archive from mongodump-2024-02-06-14-22.archive.bson
2024-02-06T23:02:19.297+0000	finished restoring test.mongodump-2024-02-06-14-22.archive (0 documents, 0 failures)
2024-02-06T23:02:19.297+0000	Failed: test.mongodump-2024-02-06-14-22.archive: error restoring from mongodump-2024-02-06-14-22.archive.bson: reading bson input: invalid BSONSize: -2120621459 bytes is less than 5 bytes
2024-02-06T23:02:19.297+0000	0 document(s) restored successfully. 0 document(s) failed to restore.
```

**Reply by exander77** [reply]
Ok, this works:
```
root@75566d3c2072:/data/backup# mongorestore --archive=mongodump-2024-02-06-14-22.archive      
```

**Reply by exander77** [reply]
This seems to work, mongorestore the archive, then dbexport, and created json can be imported.
